### PR TITLE
build: enable reuse of forks during unit tests

### PR DESCRIPTION
**Pull request by miguelatvaadin**
Several changes which enable reusing forks during the surefire (unit tests) phase. As a result, a new jvm is not started for each unit test class and a 5 minute build time gain is experienced for the full build.

Fixes: #9885

### CODE REVIEW

File: flow-server/src/test/java/com/vaadin/flow/server/StaticFileServerTest.java [hunk 1]
```diff
@@ -91,6 +93,12 @@ protected URLConnection openConnection(URL u) throws IOException {
 
     }
 
+    @BeforeClass
+    public static void beforeClass() {
```
Comment (denis-anisimov) [inline flow-server/src/test/java/com/vaadin/flow/server/StaticFileServerTest.java:97]: It's executed once for this test class.
What does this solve ?
I would expect this call in `@Before` instead `@BeforeClass` so it's not clear what does this do.

File: pom.xml [hunk 2]
```diff
@@ -491,7 +491,9 @@
                     <version>${maven.surefire.plugin.version}</version>
                     <configuration>
                         <trimStackTrace>false</trimStackTrace>
+                        <!--
```
Comment (denis-anisimov) [inline pom.xml:494]: Comments should either have explanation or removed.

**Review by denis-anisimov — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (miguelatvaadin) [reply pom.xml:494]: no problem. I will definitely remove it. Thanks!

File: flow-server/src/test/java/com/vaadin/flow/server/StaticFileServerTest.java (hunk 1 above)
Comment (miguelatvaadin) [reply flow-server/src/test/java/com/vaadin/flow/server/StaticFileServerTest.java:97]: This call solves an issue which arises when reusing forks. For some reason some trash is left there by a previous test from another class and, as a result, one of these tests fails so some clearance must be done. It was not a problem when a new jvm was instantiated for each test class, but now it is made visible.

`CurrentInstance.clearAll();` only needs to be called once for the whole class, before all these tests run. That's why I placed it in the `@BeforeClass`.

Comment (denis-anisimov) [reply flow-server/src/test/java/com/vaadin/flow/server/StaticFileServerTest.java:97]: Alright, thanks for explanation.

**Review by denis-anisimov — APPROVED** [review]

**Pull request merged**